core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -543,54 +598,71 @@ constructor(
-                                when (onVideoRecordEvent.error) {
-                                    ERROR_NONE ->
-                                        onVideoRecord(
-                                            CameraUseCase.OnVideoRecordEvent.OnVideoRecorded
-                                        )
-
-                                    else ->
-                                        onVideoRecord(
-                                            CameraUseCase.OnVideoRecordEvent.OnVideoRecordError
-                                        )
+                    .start(callbackExecutor) { onVideoRecordEvent ->
+                        run {
+                            Log.d(TAG, onVideoRecordEvent.toString())
+                            when (onVideoRecordEvent) {
+                                is VideoRecordEvent.Finalize -> {
+                                    torchDisableDeferred?.complete(Unit)
+                                    when (onVideoRecordEvent.error) {
+                                        ERROR_NONE ->
+                                            onVideoRecord(
+                                                CameraUseCase.OnVideoRecordEvent.OnVideoRecorded(
+                                                    onVideoRecordEvent.outputResults.outputUri
+                                                )
+                                            )
+
+                                        else ->
+                                            onVideoRecord(
+                                                CameraUseCase.OnVideoRecordEvent.OnVideoRecordError
+                                            )
+                                    }
                                 }
-                            }
 
-                            is VideoRecordEvent.Status -> {
-                                onVideoRecord(
-                                    CameraUseCase.OnVideoRecordEvent.OnVideoRecordStatus(
-                                        onVideoRecordEvent.recordingStats.audioStats.audioAmplitude
+                                is VideoRecordEvent.Status -> {
+                                    onVideoRecord(
+                                        CameraUseCase.OnVideoRecordEvent.OnVideoRecordStatus(
+                                            onVideoRecordEvent.recordingStats.audioStats
+                                                .audioAmplitude
+                                        )
                                     )
-                                )
+                                }
                             }
                         }
                     }
-                }
-        currentSettings.value?.audioMuted?.let { recording?.mute(it) }
-    }
+            currentSettings.value?.audioMuted?.let { recording.mute(it) }
+        } catch (e: Exception) {
+            torchDisableDeferred?.complete(Unit)
+            throw e
+        }
```

Comment:
nit: just us a `finally` block here.

Reply:
Oh, the normal path to disable the torch is when the Finalize event is received. This place is to avoid any exception is thrown when a recording can't be started. Put it in finally block will disable the torch too early.